<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ImageSelector</class>
 <widget class="QDialog" name="ImageSelector">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>524</width>
    <height>453</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Image selector</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="1" column="2">
    <widget class="QTreeWidget" name="rolesTree">
     <property name="statusTip">
      <string>Role tree</string>
     </property>
     <property name="whatsThis">
      <string>This is tree of associated and not associated roles (aliases) for images. Note that each role must be linked with one image before table export. However you can use the workspace even although not all roles are reserved yet</string>
     </property>
     <property name="accessibleName">
      <string/>
     </property>
     <property name="styleSheet">
      <string notr="true">QTreeWidget {
	background-color: rgba(80, 200, 0, 0.1);	
}
</string>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Plain</enum>
     </property>
     <property name="alternatingRowColors">
      <bool>false</bool>
     </property>
     <property name="selectionMode">
      <enum>QAbstractItemView::ExtendedSelection</enum>
     </property>
     <property name="selectionBehavior">
      <enum>QAbstractItemView::SelectRows</enum>
     </property>
     <property name="uniformRowHeights">
      <bool>true</bool>
     </property>
     <property name="animated">
      <bool>false</bool>
     </property>
     <property name="columnCount">
      <number>1</number>
     </property>
     <attribute name="headerVisible">
      <bool>false</bool>
     </attribute>
     <attribute name="headerStretchLastSection">
      <bool>true</bool>
     </attribute>
     <column>
      <property name="text">
       <string>roles</string>
      </property>
     </column>
     <item>
      <property name="text">
       <string>Linked</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Unlinked</string>
      </property>
     </item>
    </widget>
   </item>
   <item row="3" column="2">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
   <item row="0" column="0" rowspan="4">
    <widget class="QWidget" name="buttonWrapper" native="true">
     <property name="minimumSize">
      <size>
       <width>200</width>
       <height>0</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>200</width>
       <height>16777215</height>
      </size>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QPushButton" name="selectDirectoryButton">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>30</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>30</height>
         </size>
        </property>
        <property name="statusTip">
         <string>Select directory with images to be load</string>
        </property>
        <property name="whatsThis">
         <string>Click this button to evoke dialog window for selecting directory with PNG, JPG or BMP images to be loaded</string>
        </property>
        <property name="text">
         <string>Select directory</string>
        </property>
        <property name="icon">
         <iconset resource="icons.qrc">
          <normaloff>:/dialog_selector_icons/dialog_icons/image_folder_icon.png</normaloff>:/dialog_selector_icons/dialog_icons/image_folder_icon.png</iconset>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="selectImageButton">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>30</height>
         </size>
        </property>
        <property name="statusTip">
         <string>Select single image or few images to load</string>
        </property>
        <property name="whatsThis">
         <string>Click this button to evoke dialog window for direct selection of single image, or multiple set of images</string>
        </property>
        <property name="text">
         <string>Select image(s)</string>
        </property>
        <property name="icon">
         <iconset resource="icons.qrc">
          <normaloff>:/dialog_selector_icons/dialog_icons/images_icon.png</normaloff>:/dialog_selector_icons/dialog_icons/images_icon.png</iconset>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="removeSelectedButton">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>30</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>30</height>
         </size>
        </property>
        <property name="statusTip">
         <string>Remove selected items in image list</string>
        </property>
        <property name="whatsThis">
         <string>Click this button if you wish to delete selected items in imported images list</string>
        </property>
        <property name="text">
         <string>Remove selected</string>
        </property>
        <property name="icon">
         <iconset resource="icons.qrc">
          <normaloff>:/dialog_selector_icons/dialog_icons/images_remove_icon.png</normaloff>:/dialog_selector_icons/dialog_icons/images_remove_icon.png</iconset>
        </property>
       </widget>
      </item>
      <item>
       <widget class="Line" name="loadButtonsSeparator">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="linkButton">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>35</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>35</height>
         </size>
        </property>
        <property name="statusTip">
         <string>Links the selected image with selected role</string>
        </property>
        <property name="whatsThis">
         <string>If single image is selected in image list and appropriate role is highlighted, image will be associated with selected role and become hidden in images list</string>
        </property>
        <property name="text">
         <string>Link</string>
        </property>
        <property name="icon">
         <iconset resource="icons.qrc">
          <normaloff>:/dialog_selector_icons/dialog_icons/link_icon.png</normaloff>:/dialog_selector_icons/dialog_icons/link_icon.png</iconset>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="unlinkButton">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>35</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>35</height>
         </size>
        </property>
        <property name="statusTip">
         <string>Unlinks selected role with currently associated image</string>
        </property>
        <property name="whatsThis">
         <string>If a role with associated image is selected, click this button to break up the connection between them. Unlinked image will appear back in image list</string>
        </property>
        <property name="text">
         <string>Unlink</string>
        </property>
        <property name="icon">
         <iconset resource="icons.qrc">
          <normaloff>:/dialog_selector_icons/dialog_icons/unlink_icon.png</normaloff>:/dialog_selector_icons/dialog_icons/unlink_icon.png</iconset>
        </property>
       </widget>
      </item>
      <item>
       <widget class="Line" name="linkButtonsSeparator">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="previewLabel">
        <property name="minimumSize">
         <size>
          <width>180</width>
          <height>175</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>180</width>
          <height>175</height>
         </size>
        </property>
        <property name="styleSheet">
         <string notr="true">	border: 2px solid black; 
	color: red;
	font-size: 20px;
	font-weight: bold;</string>
        </property>
        <property name="text">
         <string>no preview</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item>
       <widget class="Line" name="previewSeparator">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="incrementalSwitchCheckbox">
        <property name="statusTip">
         <string>Incremental switching option on/off</string>
        </property>
        <property name="whatsThis">
         <string>This option allows to increment role item pointer to the next item if exists after association current role with image</string>
        </property>
        <property name="layoutDirection">
         <enum>Qt::LeftToRight</enum>
        </property>
        <property name="text">
         <string>Hold last item index</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="2" column="2">
    <widget class="QListWidget" name="importedImagesList">
     <property name="statusTip">
      <string>Image list</string>
     </property>
     <property name="whatsThis">
      <string>This list contains all images registered by program internally</string>
     </property>
     <property name="styleSheet">
      <string notr="true">QListWidget {
	background-color: rgba(80, 200, 0, 0.1);	
}</string>
     </property>
     <property name="selectionMode">
      <enum>QAbstractItemView::ExtendedSelection</enum>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="icons.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>ImageSelector</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>ImageSelector</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
